### PR TITLE
2fa new domain

### LINKED ISSUE #5638: Second factor login on new login domain
As a user with a second factor for [mail.tutanota.com](http://mail.tutanota.com/), I want to know how I can use it for [tuta.com](http://tuta.com/) in the future so I can migrate fully to the new domain and not worry about the old one disappearing.

## AC
* When I log in on app.tuta.com and there are no second factors for tuta.com, a message is shown (with a link) that the login must be done on the mail.tutanota.com or confirmed on another device
*  if the user has a totp, we just show the code input and a message that the hardware key is not registered for this domain
*  After I log in on app.tuta.com and there are no hardware second factors for tuta.com but hardware second factors for mail.tutanota.com, a message is shown (how?) that the key must be added to this domain.
* for this message, totp is ignored

## Tasks
 - [x] Check if the existing message about second factor domains fits and adjust if it doesn't
 - [x] Implement post-login check for U2F keys for the new domain
 - [x] Make sure that apps also pick the right domain or show the message
   - [x] desktop
   - [x] android
   - [x] ios

## Test Notes
- [ ] Have a key on old domain, try to log in on new domain
    - [ ] Login works on mobile apps and desktop
    - [ ] Web will show dialog with instructions
- [ ] Have a key on new domain, try to log in on old domain. It does not work. 
- [ ] If you have keys for both old and new domain, they new domain is preferred. (Log in on app and check the url that is brought up during the 2fa dialog.)
- [ ] Whitelabel logins still work for web and mobile apps.
- [ ] When sharing giftcard, url has correct domain. Check web and at least one mobile app.
- [ ] Refer a friend link has correct domain. Check web and at least one mobile app.

### PR DESCRIPTION
**Pull request by charlag**
Needs a bunch of testing on different domains and also probably needs some fixes on mobile but otherwise it's done

close #5638

### CODE REVIEW

File: src/api/common/Env.ts [hunk 1]
```diff
@@ -186,9 +150,25 @@ export function isOfflineStorageAvailable(): boolean {
 	return !isBrowser()
 }
 
-export function assertOfflineStorageAvailable() {
-	if (!isOfflineStorageAvailable()) {
-		throw new Error("Offline storage is not available")
+export class DomainConfigProvider {
```
Comment (ganthern) [inline src/api/common/Env.ts:153]: i think this should be a separate file, it  also feels from the usages that i could wrap our getWebRoot() and getApiUrl() and similar functions and hide the domain config completely? Basically make this an abstraction for anytime anyone wants to know something that depends on the current origin

File: src/api/main/UserController.ts [hunk 2]
```diff
@@ -292,9 +293,10 @@ export class UserController {
 				}
 			} else {
 				// Fall back to sync XHR if
```
Comment (ganthern) [inline src/api/main/UserController.ts:295]: what's this about

File: buildSrc/DomainConfigs.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
+/** @type DomainConfigMap */
+export const domainConfigs = {
```
[the remark this answers is not in the document]
Comment (charlag) [reply buildSrc/DomainConfigs.js:2]: yeah that would make sense. We have more things like that across the app.

File: src/api/common/Env.ts (hunk 1 above)
Comment (charlag) [reply src/api/common/Env.ts:153]: The abstraction is that we don't touch `DomainConfig` directly but always use the accessors?
I didn't know how to do things without building too many abstractions

File: src/api/main/UserController.ts (hunk 2 above)
Comment (ganthern) [reply src/api/main/UserController.ts:295]: yes

**Review by ganthern — APPROVED** [review]
awesome

**Pull request merged**